network/network.go

```diff
@@ -310,25 +310,40 @@ func NewNetwork(
 	return n, nil
 }
 
-func (n *network) Send(msg message.OutboundMessage, nodeIDs set.Set[ids.NodeID], subnetID ids.ID, allower subnets.Allower) set.Set[ids.NodeID] {
-	peers := n.getPeers(nodeIDs, subnetID, allower)
-	n.peerConfig.Metrics.MultipleSendsFailed(
-		msg.Op(),
-		nodeIDs.Len()-len(peers),
-	)
-	return n.send(msg, peers)
-}
-
-func (n *network) Gossip(
+func (n *network) Send(
 	msg message.OutboundMessage,
+	config common.SendConfig,
 	subnetID ids.ID,
-	numValidatorsToSend int,
-	numNonValidatorsToSend int,
-	numPeersToSend int,
 	allower subnets.Allower,
 ) set.Set[ids.NodeID] {
-	peers := n.samplePeers(subnetID, numValidatorsToSend, numNonValidatorsToSend, numPeersToSend, allower)
-	return n.send(msg, peers)
+	namedPeers := n.getPeers(config.NodeIDs, subnetID, allower)
+	n.peerConfig.Metrics.MultipleSendsFailed(
+		msg.Op(),
+		config.NodeIDs.Len()-len(namedPeers),
+	)
+
+	var (
+		sampledPeers = n.samplePeers(config, subnetID, allower)
+		sentTo       = set.NewSet[ids.NodeID](len(namedPeers) + len(sampledPeers))
+		now          = n.peerConfig.Clock.Time()
+	)
+
+	// send to peer and update metrics
+	for _, peers := range [][]peer.Peer{namedPeers, sampledPeers} {
+		for _, peer := range peers {
+			if peer.Send(n.onCloseCtx, msg) {
```

Comment:
As noted below, we don't need to check `sentTo` because we avoid adding `namedPeers` to `sampledPeers`.